### PR TITLE
Fix build with qwlroots on latest

### PR DESCRIPTION
**Pull request by justforlxz**

### CODE REVIEW

File: src/server/qtquick/wquickcursor.cpp [hunk 1]
```diff
@@ -20,10 +20,7 @@ class WQuickCursorPrivate : public WCursorPrivate
 {
 public:
     WQuickCursorPrivate(WQuickCursor *qq);
-    ~WQuickCursorPrivate()
-    {
-        xcursor_manager->destroy();
```
Comment (wineee) [inline src/server/qtquick/wquickcursor.cpp:25]: delete xcursor_manager;
其他地方类似

File: src/server/kernel/wserver.cpp [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Copyright (C) 2023 JiDe Zhang <zhangjide@deepin.org>.
 // SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
 
+#include <QCoreApplication>
```
Comment (zccrs) [inline src/server/kernel/wserver.cpp:4]: Why?

File: src/server/kernel/wserver.cpp [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+
 #define private public
 #include <QCoreApplication>
```
Comment (zccrs) [inline src/server/kernel/wserver.cpp:7]: QCoreApplication at here

File: src/server/kernel/wxdgshell.cpp [hunk 4]
```diff
@@ -39,11 +42,11 @@ void WXdgShellPrivate::on_new_xdg_surface(wlr_xdg_surface *wlr_surface)
 {
     auto server = q_func()->server();
     // TODO: QWXdgSurface::from(wlr_surface)
-    QWXdgSurface *xdgSurface = QWXdgSurface::from(wlr_surface->surface);
+    QWSurface *xdgSurface = QWSurface::from(wlr_surface->surface);
```
Comment (zccrs) [inline src/server/kernel/wxdgshell.cpp:45]: Why change to QWSurface?

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**